fuzz/fuzz_targets/decompress_lzma_stream.rs

```diff
@@ -0,0 +1,40 @@
+#![no_main]
+#[macro_use]
+extern crate libfuzzer_sys;
+
+use lzma_rs::error::Result;
+use std::io::Write;
+
+fn decode_lzma(compressed: &[u8]) -> Result<Vec<u8>> {
+    let mut bf = std::io::Cursor::new(compressed);
+    let mut decomp: Vec<u8> = Vec::new();
+    lzma_rs::lzma_decompress(&mut bf, &mut decomp)?;
+    Ok(decomp)
+}
+
+fn decode_lzma_stream(compressed: &[u8], chunk_size: usize) -> Result<Vec<u8>> {
+    let mut stream = lzma_rs::decompress::Stream::new(Vec::new());
+    for chunk in compressed.chunks(chunk_size) {
+        stream.write_all(chunk).unwrap();
+    }
+    let decomp = stream.finish().unwrap();
+    Ok(decomp)
+}
+
+fuzz_target!(|input: &[u8]| {
+    let chunk_size = 2;
```

Comment:
Here I think it would be interesting to fuzz various chunk sizes depending on the input. I'd suggest defining the chunk size as the first byte of the input, and let the rest of the input be the actual data. Something like the following:

```rust
let (chunk_size, data) = input.split_at(1);
let chunk_size = chunk_size[0] as usize;
```

This won't work when the input is empty, but in that case we can simply return directly.